src/Controller/FilesController.php

```diff
@@ -78,16 +81,25 @@ public function fileAction(Request $request, string $file): BinaryFileResponse
         $allowLogin = false;
         $allowAccess = false;
 
+        // Get the current user
+        $user = $this->security->getUser();
+
+        // Check if this is the home directory
+        $isHomeDir = $user instanceof FrontendUser && $user->homeDir === $filesModel->uuid;
+
         do {
             // Only check for folders and when member groups have been set
-            if ('folder' === $filesModel->type && null !== $filesModel->groups) {
+            // or access to member home directory
+            if ('folder' === $filesModel->type
+                && (null !== $filesModel->groups
+                    || ($user->accessHomeDir && $user->homeDir))) {
                 $allowLogin = true;
 
                 // Set the model to protected on the fly
                 $filesModel->protected = true;
 
                 // Check access
-                if (Controller::isVisibleElement($filesModel)) {
+                if (($canAccessHomeDir && $isHomeDir) || (Controller::isVisibleElement($filesModel))) {
```

Comment:
The `$canAccessHomeDir` variable is never defined and thus this is always false.

See https://github.com/fritzmg/contao-file-access/pull/10#discussion_r790512050 - `$canAccessHomeDir` should be defined before the loop, `$isHomeDir` needs to be defined within the loop, but probably before the first condition. The condition will need to be adjusted in general though.